### PR TITLE
Add experimental pokemon upgrade (power-up) logic

### PR DESCRIPTION
**Pull request by Anakin5**
Add experimental pokemon upgrade (power-up) logic
Add upgrade cost data file

### CODE REVIEW

**Comment by DBa2016** [comment]
Fixes #4913

File: data/pokemon_upgrade_cost.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
+[
+[1, 200],
+[1, 200],
```
Comment (mjmadsen) [inline data/pokemon_upgrade_cost.json]: How come this doesn't go [1,200],[2,200] and so forth? Sorry, there a whole lot of edits, still working through it all.

Comment (Anakin5) [inline data/pokemon_upgrade_cost.json]: According to my data, the price change every 4 pokemon level. I am not sure data is correct though, need to double check.

Comment (mjmadsen) [inline data/pokemon_upgrade_cost.json]: As far as I know, that's true. So we're using the entry index as our level and the first index in the entry as our level "group"?

Comment (Anakin5) [inline data/pokemon_upgrade_cost.json]: Index in the list is `pokemon.level * 2 - 1`. First value is candy count, second is stardust.

Comment (mjmadsen) [inline data/pokemon_upgrade_cost.json]: Oh! Seems obvious now that you explained it :)

**Comment by mjmadsen** [comment]
CI failed :(

**Comment by Anakin5** [comment]
what is the problem ?

**Comment by mjmadsen** [comment]
======================================================================
FAIL: test_levels_to_cpm (tests.inventory_test.InventoryTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/PokemonGoF/PokemonGo-Bot/tests/inventory_test.py", line 165, in test_levels_to_cpm
    self.assertEqual(len(l2c.STATIC_DATA), 79)
AssertionError: 80 != 79

The commit before the last passed, for some reason the last commit failed.

**Comment by Anakin5** [comment]
Yeah I added one level to levels_to_cpm file. Not sure why it is expecting 79 though since there is 80 Pokemon levels.

**Comment by rawgni** [comment]
@Anakin5 since all pokemon starts at lv1. it's 79 to get to 80.

**Comment by Anakin5** [comment]
This file is used to get Pokemon level (from their cp multiplier), which goes from 1 included to 80 included.
And according to other sources, the last level was indeed missing.

**Comment by mjmadsen** [comment]
@Anakin5 so it's the test that is outdated?

**Comment by Anakin5** [comment]
I don't really know how was that file created in the first place. I am not sure why levels are going from 1 to 40 by step of 0.5, while they should go from 1 to 80 by step of 1.
I think the last line might have been missed originally because of that conversion. Test should count 80. We can verify that by fully powering up a Pokemon and see it's coming is missing.

**Comment by mjmadsen** [comment]
Could you just remove the last line for now. It doesn't really matter right now. None of us are level 40 and finishing off a perfect pokemon. Merge dev/rebase whenever you're done and I'll push.

Side note, it does seem a bit bizarre levels are stored like you mentioned. Do you plan to look into it?

**Comment by Anakin5** [comment]
I changed the assert

**Comment by mjmadsen** [comment]
:+1: 

[Approved with PullApprove](https://pullapprove.com/PokemonGoF/PokemonGo-Bot/pull-request/4938/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=PokemonGoF/PokemonGo-Bot)

**Comment by mjmadsen** [comment]
:+1:

**Pull request merged**